WordPress/src/main/java/org/wordpress/android/ui/media/MediaPickerActivity.java

```diff
@@ -453,30 +457,48 @@ public void onAnimationRepeat(Animation animation) {
      */
     public class MediaPickerAdapter extends FragmentPagerAdapter {
         private class MediaPicker {
+            public String loadingText;
+            public String errorText;
+            public String emptyText;
             public String pickerTitle;
             public ArrayList<MediaSource> mediaSources;
 
-            public MediaPicker(String name, ArrayList<MediaSource> sources) {
+            public MediaPicker(String name, String loading, String error, String empty, ArrayList<MediaSource> sources) {
+                loadingText = loading;
+                errorText = error;
+                emptyText = empty;
                 pickerTitle = name;
                 mediaSources = sources;
             }
         }
 
-        private List<MediaPicker> mMediaPickers;
+        private final List<MediaPicker> mMediaPickers;
+        private final List<MediaPickerFragment> mFragments;
```

Comment:
Since you're keeping a list of created fragments, you may need to override `destroyItem()` to make sure removed fragments aren't retained in `mFragments`. [Here's an example](https://github.com/wordpress-mobile/WordPress-Android/blob/issue/2414-empty-media-view-squashed/WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostPagerActivity.java#L546).